src/coreclr/nativeaot/Runtime/i386/StubDispatch.asm

```diff
@@ -31,16 +31,17 @@ endm
 DEFINE_INTERFACE_DISPATCH_STUB macro entries
 
 StubName textequ @CatStr( _RhpInterfaceDispatch, entries, <@0> )
+StubAVLocation textequ @CatStr( _RhpInterfaceDispatchAVLocation, entries )
 
     StubName proc public
 
         ;; Check the instance here to catch null references. We're going to touch it again below (to cache
         ;; the MethodTable pointer), but that's after we've pushed ebx below, and taking an A/V there will
-        ;; mess up the stack trace for debugging. We also don't have a spare scratch register (eax holds
-        ;; the cache pointer and the push of ebx below is precisely so we can access a second register
-        ;; to hold the MethodTable pointer).
-        test    ecx, ecx
-        je      RhpInterfaceDispatchNullReference
+        ;; mess up the stack trace. We also don't have a spare scratch register (eax holds the cache pointer
+        ;; and the push of ebx below is precisely so we can access a second register to hold the MethodTable
+        ;; pointer).
+    ALTERNATE_ENTRY StubAVLocation
+        cmp     byte ptr [ecx], 0
```

Comment:
```suggestion
        cmp     dword ptr [ecx], 0
```

If we are comparing against a pointer I _think_ this should be `dword`.

Reply:
It doesn't necessarily matter since it's just a null check to trigger access violation. I opted to match the x64 code which also does just a byte check.

Reply:
It is `this` pointer null check. Anything will work.

The null check sequence used by the JIT is `cmp dword ptr [ecx],ecx` so it would be the most sensible one to use here as well.

https://github.com/dotnet/runtime/blob/03b5de78d91cf3c971d251a819bd78b6444838bb/src/coreclr/jit/codegenxarch.cpp#L5956